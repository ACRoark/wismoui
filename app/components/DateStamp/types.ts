export type DateStampFormat = 'auto' | 'long' | 'short';
