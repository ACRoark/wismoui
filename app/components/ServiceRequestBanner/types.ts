export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
