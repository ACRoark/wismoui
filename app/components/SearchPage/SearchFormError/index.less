@import '../../../styles/themes/index.less';

@search-form-error-cls-prefix: ~'@{dte-prefix}-search-form-error';

@desktop: ~'(min-width: 1025px)';

.@{search-form-error-cls-prefix} {
  align-items: flex-start;
  align-self: flex-start;
  background-color: @dte-red-light;
  color: @dte-red;
  display: flex;
  font-size: @dte-wismo-font-size-0;
  justify-content: space-between;
  margin-top: 30px;
  padding: 25px;

  @media @desktop {
    align-self: center;
    width: 230%;
  }

  .close-button {
    align-items: center;
    background-color: @dte-red-light;
    border: none;
    color: @dte-red;
    cursor: pointer;
    display: flex;
    height: 18px;
    justify-content: center;
    margin-top: 2px;
  }

  .icon {
    font-size: 1.875rem;
    font-weight: 600;
    margin-right: 20px;
  }

  .message {
    align-items: flex-start;
    display: flex;
  }
}
