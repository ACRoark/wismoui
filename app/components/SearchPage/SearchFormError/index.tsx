import React, { FC, ReactElement } from 'react';
import { FormattedMessage } from 'react-intl';

import useThunkDispatch from 'hooks/useThunkDispatch';
import { resetTracker } from 'store/actions';

import messages from './messages';

import './index.less';

interface ISearchFormError {
  error: string;
}

const SearchFormError: FC<ISearchFormError> = (props: ISearchFormError): ReactElement => {
  const { error } = props;

  const dispatch = useThunkDispatch();

  const handleClick = (): void => {
    dispatch(resetTracker());
  };

  const getMessage = (errorMsg: string): ReactElement => {
    switch (errorMsg) {
      case 'ORDER_INACTIVE':
        return <FormattedMessage {...messages.inactive} />;

      case 'ORDER_INVALID':
        return <FormattedMessage {...messages.invalid} />;

      case 'ORDER_NOT_FOUND':
        return <FormattedMessage {...messages.orderNotFound} />;

      case 'RECAPTCHA_FAILED':
        return <FormattedMessage {...messages.recaptchaFailed} />;

      case 'SERVER_UNAVAILABLE':
        return <FormattedMessage {...messages.unavailable} />;

      default:
        return <FormattedMessage {...messages.unexpected} />;
    }
  };

  return (
    <div className="dte-wismo-search-form-error">
      <div className="message">
        <span className="icon">!</span>
        {getMessage(error)}
      </div>
      <button className="close-button" onClick={handleClick} type="button">
        <svg width="18" height="18" viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
          <path
            d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
            stroke="currentColor"
            strokeWidth="2"
            strokeLinecap="round"
          />
        </svg>
      </button>
    </div>
  );
};

export default SearchFormError;
