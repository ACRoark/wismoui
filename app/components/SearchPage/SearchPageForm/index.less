@import '../../../styles/themes/index.less';

@search-page-form-cls-prefix: ~'@{dte-prefix}-search-page-form';

.@{search-page-form-cls-prefix} {
  display: flex;
  flex-direction: column;

  .title {
    font-size: @dte-wismo-font-size-1;
    line-height: 1.875rem;
    text-align: center;
  }

  .recaptcha {
    display: flex;
    justify-content: center;
    transform: scale(1.08);
  }

  .button {
    align-self: center;
    background-color: @dte-blue-light;
    border: none;
    color: @dte-white;
    cursor: pointer;
    font-size: @dte-wismo-font-size-1;
    font-weight: @dte-wismo-font-weight-semi-bold;
    height: 40px;
    margin-top: 30px;
    width: 142px;

    &:hover {
      background-color: @dte-blue;
    }
  }

  .button:disabled {
    background-color: @dte-gray-500;
    cursor: not-allowed;
  }

  input {
    border: 1px solid @dte-gray-400;
    height: 36px;
    outline: none;
    padding-left: 12px;
    padding-right: 12px;
    width: 100%;

    &[name='order-number'] {
      margin-bottom: 30px;
    }

    &:focus {
      border: 1px solid @dte-blue;
    }
  }

  &-input-header {
    margin-bottom: 4px;
  }

  label {
    color: @dte-gray-600;
    font-size: @dte-wismo-font-size-s;
    font-weight: @dte-wismo-font-weight-semi-bold;
    margin-top: 30px;
  }
}
