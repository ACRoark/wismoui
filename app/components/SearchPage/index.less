@import (reference) '../../styles/themes/index.less';

@search-page-cls-prefix: ~'@{dte-prefix}-search-page';

.@{search-page-cls-prefix} {
  display: flex;
  justify-content: center;

  &-container {
    border: 1px solid @dte-gray-300;
    display: flex;
    justify-content: center;
    margin: 44px 0 99px;
    max-width: 952px;
    width: 80%;

    @media @mobile {
      border: none;
      margin: 0;
      width: 100%;
    }
  }

  &-content {
    margin: 48px 0 41px;
    width: 327px;
  }
}
