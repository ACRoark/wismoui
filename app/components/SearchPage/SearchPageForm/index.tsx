import React, { FC, ReactElement, RefObject, SyntheticEvent, useEffect, useState } from 'react';
import ReCAPTCHA from 'react-google-recaptcha';
import InputMask from 'react-input-mask';
import { FormattedMessage } from 'react-intl';
import { useLocation } from 'react-router-dom';

import useConfig from 'hooks/useConfig';
import useFlags from 'hooks/useFlags';
import useThunkDispatch from 'hooks/useThunkDispatch';
import encryptionApiClient from 'services/encryptionApiClient';
import { findOrder, resetTracker } from 'store/actions';
import SearchFormError from '../SearchFormError';
import getEncodedName from './getEncodedName';
import messages from './messages';

import './index.less';

interface ISearchPageFormProps {
  error?: string;
}

const SearchPageForm: FC<ISearchPageFormProps> = (props: ISearchPageFormProps): ReactElement => {
  const { error } = props;

  const { bug2263 } = useFlags();

  const dispatch = useThunkDispatch();
  const [isDisabled, setIsDisabled] = useState(true);
  const [name, setName] = useState('');
  const [orderNumber, setOrderNumber] = useState('');
  const { search: queryParameter = '' } = useLocation();
  const recaptchaRef: RefObject<any> = React.createRef(); // tslint:disable-line
  const {
    recaptchaSitekey,
    urls: { decryptTokenApi },
  } = useConfig();
  const [recaptchaToken, setRecaptchaToken] = useState('');

  const handleDisabled = (): void =>
    name.length && orderNumber.length && recaptchaToken ? setIsDisabled(false) : setIsDisabled(true);

  const handleNameChange = (event: SyntheticEvent): void => {
    const element = event.target as HTMLInputElement;

    setName(element.value);
  };

  const handleOrderNumberChange = (event: SyntheticEvent): void => {
    const element = event.target as HTMLInputElement;

    setOrderNumber(element.value);
  };

  const handleSubmit = async (event: SyntheticEvent): Promise<void> => {
    event.preventDefault();

    dispatch(findOrder(orderNumber.replace(/\s/g, ''), getEncodedName(name), recaptchaToken, bug2263));
    setIsDisabled(true);
  };

  const recaptchaOnChange = (value: string): void => {
    setRecaptchaToken(value || '');
    dispatch(resetTracker());
  };

  useEffect((): void => {
    if (queryParameter) {
      const setDecryptedOrderNumber = async (): Promise<void> => {
        const decryptedOrderNumber = await encryptionApiClient.decryptTokenAsync(decryptTokenApi, queryParameter);

        setOrderNumber(decryptedOrderNumber);
      };

      setDecryptedOrderNumber();
    }
  }, []);

  useEffect((): void => {
    handleDisabled();
  }, [recaptchaToken, name, orderNumber]);

  useEffect((): void => {
    if (error && recaptchaRef.current) {
      recaptchaRef.current.reset();
      setRecaptchaToken('');
    }
  }, [error]);

  return (
    <form
      className="dte-wismo-search-page-form"
      id="mimotrackerguestsearch"
      name="mimotrackerguestsearch"
      onSubmit={handleSubmit}
    >
      <h1 className="title">
        <FormattedMessage {...messages.title} />
      </h1>

      <label htmlFor="name">
        <div className="dte-wismo-search-page-form-input-header">
          <FormattedMessage {...messages.name} />
        </div>
        <input name="name" type="text" value={name} onChange={handleNameChange} />
      </label>

      <label htmlFor="order-number">
        <div className="dte-wismo-search-page-form-input-header">
          <FormattedMessage {...messages.orderNumber} />
        </div>
        <InputMask
          mask="** **** ****"
          maskChar={null}
          name="order-number"
          onChange={handleOrderNumberChange}
          value={orderNumber}
        />
      </label>

      <ReCAPTCHA className="recaptcha" ref={recaptchaRef} sitekey={recaptchaSitekey} onChange={recaptchaOnChange} />

      {error && <SearchFormError error={error} />}

      <button
        className="button"
        data-track="mimo"
        data-track-detail="order tracker"
        data-track-action="click"
        data-track-sub-action="guest search"
        type="submit"
        disabled={isDisabled}
      >
        <FormattedMessage {...messages.button} />
      </button>
    </form>
  );
};

export default SearchPageForm;
