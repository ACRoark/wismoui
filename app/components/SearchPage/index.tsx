import React, { FC, ReactElement, useEffect } from 'react';
import { Helmet } from 'react-helmet';
import { Redirect } from 'react-router-dom';

import { PageTitles } from 'components/constants';
import useThunkDispatch from 'hooks/useThunkDispatch';
import useTracker from 'hooks/useTracker';
import { userIsAuthenticated } from 'security/utils';
import { resetTracker, sendPageView } from 'store/actions';
import SearchPageForm from './SearchPageForm';

import './index.less';

const SearchPage: FC = (): ReactElement => {
  // Do not let authenticated users access this page
  if (userIsAuthenticated()) {
    return <Redirect to="/" />;
  }

  const dispatch = useThunkDispatch();

  useEffect((): void => {
    dispatch(sendPageView(PageTitles.SEARCH_PAGE));

    dispatch(resetTracker());
  }, []);

  const { error, loading, order } = useTracker();

  // TODO: Handle loading

  if (!error && !loading && order) {
    return <Redirect to={`/orders/${order.orderNumber}`} />;
  }

  return (
    <div className="dte-wismo-search-page">
      <Helmet>
        <title>{PageTitles.SEARCH_PAGE}</title>
      </Helmet>
      <div className="dte-wismo-search-page-container">
        <div className="dte-wismo-search-page-content">
          <SearchPageForm error={error} />
        </div>
      </div>
    </div>
  );
};

export default SearchPage;
