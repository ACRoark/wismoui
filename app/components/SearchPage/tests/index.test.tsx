import { RouterState } from 'connected-react-router';
import { mock } from 'jest-mock-extended';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import { Route } from 'react-router-dom';

import { SUPPORTED_LOCALES } from 'locales';
import { userIsAuthenticated } from 'security/utils';
import { createMockStore, createSnapshotWithIntl  } from 'testing/utils';
import { IApplicationConfig, IApplicationRootState, IAuthenticationState, ISearchState, IStartupState } from 'types';
import SearchPage from '..';

jest.mock('security/utils');

const defaultState: IApplicationRootState = {
  auth: mock<IAuthenticationState>(),
  config: mock<IApplicationConfig>(),
  router: mock<RouterState>(),
  search: mock<ISearchState>(),
  startup: mock<IStartupState>(),
  tracker: {
    error: undefined,
    loading: false,
    order: undefined,
  },
};

const mockAuth = userIsAuthenticated as jest.MockedFunction<typeof userIsAuthenticated>;

const testSearchPageSnapshot = (initialState: IApplicationRootState = defaultState, language: string, url: string): void => {
  const store = createMockStore(initialState);

  const tree = createSnapshotWithIntl(
    <MemoryRouter initialEntries={[url]}>
      <Route path="search">
        <Provider store={store}>
          <SearchPage />
        </Provider>
      </Route>
    </MemoryRouter>, language);

  expect(tree).toMatchSnapshot();
};
describe('<SearchPage />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  mockAuth.mockReturnValue(false);

  languages.forEach((language: string): void => {
    it(`should render correctly when there is no error and the language is ${language}`, (): void => {
      testSearchPageSnapshot(defaultState, language, 'search');
    });
  });
});
