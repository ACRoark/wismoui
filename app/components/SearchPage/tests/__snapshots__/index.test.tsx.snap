// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> should render correctly when there is no error and the language is en 1`] = `
<div
  className="dte-wismo-search-page"
>
  <div
    className="dte-wismo-search-page-container"
  >
    <div
      className="dte-wismo-search-page-content"
    >
      <form
        className="dte-wismo-search-page-form"
        id="mimotrackerguestsearch"
        name="mimotrackerguestsearch"
        onSubmit={[Function]}
      >
        <h1
          className="title"
        >
          To check the status of your order, please enter your information below.
        </h1>
        <label
          htmlFor="name"
        >
          <div
            className="dte-wismo-search-page-form-input-header"
          >
            Last Name or Business Name
          </div>
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label
          htmlFor="order-number"
        >
          <div
            className="dte-wismo-search-page-form-input-header"
          >
            Order Number
          </div>
          <input
            name="order-number"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            value=""
          />
        </label>
        <div
          className="recaptcha"
        />
        <button
          className="button"
          data-track="mimo"
          data-track-action="click"
          data-track-detail="order tracker"
          data-track-sub-action="guest search"
          disabled={true}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<SearchPage /> should render correctly when there is no error and the language is es 1`] = `
<div
  className="dte-wismo-search-page"
>
  <div
    className="dte-wismo-search-page-container"
  >
    <div
      className="dte-wismo-search-page-content"
    >
      <form
        className="dte-wismo-search-page-form"
        id="mimotrackerguestsearch"
        name="mimotrackerguestsearch"
        onSubmit={[Function]}
      >
        <h1
          className="title"
        >
          Para verificar el estado de su pedido, ingrese su información a continuación.
        </h1>
        <label
          htmlFor="name"
        >
          <div
            className="dte-wismo-search-page-form-input-header"
          >
            Apellido o nombre comercial
          </div>
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label
          htmlFor="order-number"
        >
          <div
            className="dte-wismo-search-page-form-input-header"
          >
            Número de orden
          </div>
          <input
            name="order-number"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            value=""
          />
        </label>
        <div
          className="recaptcha"
        />
        <button
          className="button"
          data-track="mimo"
          data-track-action="click"
          data-track-detail="order tracker"
          data-track-sub-action="guest search"
          disabled={true}
          type="submit"
        >
          Enviar
        </button>
      </form>
    </div>
  </div>
</div>
`;
